Classes/Common/DocumentCacheManager.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * (c) Kitodo. Key to digital objects e.V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo and TYPO3 projects.
+ *
+ * @license GNU General Public License version 3 or later.
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Kitodo\Dlf\Common;
+
+use TYPO3\CMS\Core\Cache\CacheManager;
+use TYPO3\CMS\Core\Cache\Frontend\FrontendInterface;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * DocumentCacheManager class for the 'dlf' extension
+ *
+ * @package TYPO3
+ * @subpackage dlf
+ *
+ * @access public
+ */
+class DocumentCacheManager
+{
+    /**
+     * @var FrontendInterface
+     */
+    protected $cache;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->cache = GeneralUtility::makeInstance(CacheManager::class)->getCache('tx_dlf_doc');
+    }
+
+    /**
+     * Get document instance from cache or false if not found.
+     *
+     * @access public
+     *
+     * @param string $location
+     *
+     * @return AbstractDocument|false
+     */
+    public function get(string $location)
+    {
+        return $this->cache->get($this->getIdentifier($location));
```

Comment:
I am afraid that here is reason for issue #1515. If the cache contains an instance of `MetsDocument`, the function returns a `MetsDocument` without setting `$useGroupsConfiguration` which is part of the base class `AbstractDocument`. I expect that other variables from `AbstractDocument` are missing, too.

Like this the caching makes Kitodo.Presentation unusable.